### DISCUSSION TITLE
Error: Please install pg package manually

### DISCUSSION THREAD

**Discussion by DaniyalKukda** [vercel/community#234] [Help] [answered]
I have deployed my application on vercel using githubActions all the dependency installations and build are successfully done but when i hit an api that communicates with my postgres db the deployment logs throw this error **Error: Please install pg package manually**

I have a pg package in my dependencies in package.json its working fine locally on my machine but after deployment i got this error

**Comment by Pieparker** [comment]
Hey there! Hard to comment on specific packages but a [quick Google search](https://claudelalyre.wordpress.com/2019/08/07/webpack-and-error-please-install-pg-package-manually/) did find this and there's some steps to try there.

Failing that, please share a commit SHA or the URL to a deployment and we can double-check to see if there is any more information we can provide.

**Reply by DaniyalKukda** [reply]
Deployed URl : https://inventory-system-nextjs.vercel.app/login

![image](https://user-images.githubusercontent.com/46843936/150995748-1384bc6f-9377-40ee-ae22-f4ff07bd3ae1.png)


this is the error all the node_modules installed successfully but when i hit any request that communicate with Database facing this error

**Reply by Pieparker** [reply]
Thanks for this! You may want to give our [serverless function failure](https://vercel.com/support/articles/why-does-my-serverless-function-work-locally-but-not-when-deployed) guide a read. Lots of useful tidbits about why errors may occur on Vercel versus locally!

**Comment by kuriel-trivu** [comment]
why is this answered? i dont see any solution here, any DB package failure looks intentionally made by Vercel, to aviod custom servers.

![image](https://user-images.githubusercontent.com/46456105/155644450-f64e9fca-5515-4455-bcaa-07d083556bd1.png)

**Reply by SoozaV** [reply]
I had to deploy my app using fly.io, exactly the same code, no other config needed.

**Reply by AshCript** [reply]
> why is this answered? i dont see any solution here, any DB package failure looks intentionally made by Vercel, to aviod custom servers.
> 
> ![image](https://user-images.githubusercontent.com/46456105/155644450-f64e9fca-5515-4455-bcaa-07d083556bd1.png)

I've tried to resolve it from the vercel GUI build command input field, by adding the script below, but it doesn't work... 🥴
```sh
npm i -g pg && npm run vercel-build
```

**Reply by AshCript** [reply]
Resolved 🎉.
Finally, I've resolved it, not by installing pg globally with npm, but by adding the dialectModule option into my database configuration as below :

```js
export const dbConfig = {
  username: process.env.DB_USERNAME,
  password: process.env.DB_PASSWORD,
  database: process.env.DB_NAME,
  host: process.env.DB_HOSTNAME,
  port: process.env.DB_PORT,
  dialect: 'postgres',
  dialectModule: pg, // I've added this.
  timezone: process.env.TZ,
  define: {
    charset: 'utf8mb4',
    collate: 'utf8mb4_general_ci',
  },
  logging: false,
};
```

**Reply by Sekramo13** [reply]
Can confirm this fixed the problem for me as well.

I'm using Sequalize with Postgres and the app router

```
import "dotenv/config"
import Sequelize from "sequelize"
import pg from "pg"
console.log("process.env: ", process.env)
export const sequelize = new Sequelize(
  process.env.DB_NAME,
  process.env.DB_USERNAME,
  process.env.DB_PASSWORD,
  {
    host: "localhost",
    dialect: "postgres",
    dialectModule: pg,
  }
)
```

**Reply by beterrabaA** [reply]
u save me

**Comment by aptarmy** [comment]
I'm facing the same problem. Are there any updates on this by far?

**Comment by Redm4x** [comment, marked as answer]
I had the same issue. Working locally, but not on vercel. What fixed it for me was to import pg manually with: `require('pg')`

**Reply by rommyarb** [reply]
Hi. From where/which file should I call this `require('pg')`? Thanks!

**Reply by lequanghuylc** [reply]
> Hi. From where/which file should I call this `require('pg')`? Thanks!

Try this:
```
const sequelize = new Sequelize('postgres://admin:admin@localhost:5432/mydb', {
  dialectModule: require('pg')
});
```

**Reply by maniac-tech** [reply]
> > Hi. From where/which file should I call this `require('pg')`? Thanks!
> 
> Try this:
> 
> ```
> const sequelize = new Sequelize('postgres://admin:admin@localhost:5432/mydb', {
>   dialectModule: require('pg')
> });
> ```

I tried your suggestion @lequanghuylc , but I can still see this happening.

**Reply by mikhailkusuma** [reply]
thanks, this work for me :)
